src/main/java/com/peerrequest/app/api/ReviewsController.java

```diff
@@ -0,0 +1,96 @@
+package com.peerrequest.app.api;
+
+import com.peerrequest.app.model.*;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public class ReviewsController {
+
+    @GetMapping(value = "/categories/{categoryId}/entries/{entryId}/reviews",
+            produces = "application/json")
+    public List<Review> getReviews(@PathVariable final int categoryId, @PathVariable final int entryId) {
+        return List.of(mockReview());
+    }
+
+    @GetMapping(value = "/categories/{categoryId}/entries/{entryId}/reviews/{reviewId}",
+            produces = "application/json")
+    public Review getReview(@PathVariable final int categoryId,
+                            @PathVariable final int entryId,
+                            @PathVariable final int reviewId) {
+        return mockReview();
+    }
+
+    @GetMapping(value = "/categories/{categoryId}/entries/{entryId}/reviews/{reviewId}/document",
+            produces = "application/json")
+    public Document getReviewDocument(@PathVariable final int categoryId,
+                                      @PathVariable final int entryId,
+                                      @PathVariable final int reviewId) {
+        return mockReview().getReviewDocument();
+    }
+
+    @PostMapping(value = "/categories/{categoryId}/entries/{entryId}/reviews",
+            consumes = "application/json",
+            produces = "application/json")
+    public Review postReview(@PathVariable final int categoryId,
+                               @PathVariable final int entryId,
+                               @RequestBody final Review review) {
+        return mockReview();
+    }
+
+    @PutMapping(value = "/categories/{categoryId}/entries/{entryId}/reviews/{reviewId}/document",
+            consumes = "application/json",
+            produces = "application/json")
+    public Document putReviewDocument(@PathVariable final int categoryId,
+                                    @PathVariable final int entryId,
+                                    @PathVariable final int reviewId,
+                                    @RequestBody final Document document) {
+        return mockReview().getReviewDocument();
+    }
+
+
+    @PatchMapping(value = "categories/{categoryId}/entries/{entryId}/reviews/{reviewId}",
+            consumes = "application/json",
+            produces = "application/json")
+    public Review patchReview(@PathVariable final int categoryId,
+                                      @PathVariable final int entryId,
+                                      @PathVariable final int reviewId,
+                                      @RequestBody final Review.ReviewUpdater reviewUpdater) {
+        Review updatedReview =  mockReview();
+        updatedReview.setScore(5);
+        updatedReview.setSummary("This is a brand new summary.");
+        return updatedReview;
```

Comment:
doesn't update the hardcoded review. Maybe updating the hardcoded review is not needed. Evaluate by yourself.